docs/articles/documentation/guides/advanced-guides/test-https-features-and-http2-websites.md

```diff
@@ -23,21 +25,95 @@ If the tested page does not use HTTPS-specific features (like
+
+    ```sh
+    openssl genrsa -des3 -out myCA.key 2048
+    ```
+
+    You will be prompted for a non-empty passphrase. Take note of it.  
+2. Generate a root certificate:
+
+    ```sh
+    openssl req -x509 -new -nodes -key myCA.key -sha256 -days 1825 -out myCA.pem
+    ```
+
+    You will be prompted for additional information, skip the the questions by pressing Enter.
+3. Create a CA-signed domain certificate and a `.csr` sign request:
+
+    ```sh
+    openssl genrsa -out testingdomain.key 2048
+    openssl req -new -key testingdomain.key -out testingdomain.csr
+    ```
+
+    You can skip all the questions by pressing Enter, but you might want to set a recognizable CN (Common Name) so that you can easily find this certificate in a list of others later.  
+4. Create a config file `testdomain.ext` with the following contents:
+  
+    ```sh
+    authorityKeyIdentifier=keyid,issuer
+    basicConstraints=CA:FALSE
+    keyUsage = digitalSignature, nonRepudiation, keyEncipherment, dataEncipherment
+    subjectAltName = @alt_names
+
+    [alt_names]
+    DNS.1 = localhost
+    ```
+
+5. Now create the certificate using our `.csr`, the CA private key (`myCA.key`), the CA certificate (`myCA.pem`), and a config file:
+
+    ```sh
+    openssl x509 -req -in testingdomain.csr -CA myCA.pem -CAkey myCA.key -CAcreateserial -out testingdomain.crt -days 825 -sha256 -extfile testdomain.ext
+    ```
+
+6. Finally, export the certificate to PKCS#12 `.pfx` format:
+
+    ```sh
+    openssl pkcs12 -export -out testingdomain.pfx -inkey testingdomain.key -in testingdomain.crt -certfile myCA.pem
+    ```
+
+In **PowerShell**, [install OpenSSL](https://adamtheautomator.com/openssl-powershell/#Installing_OpenSSL_using_Chocolatey) and [update environment variables](https://adamtheautomator.com/openssl-powershell/#Update_PowerShell_Profile_Environment_Variables). Afterwards, proceed with the instructions [above](#generate-a-certificate-from-the-command-line).
+
+##### Run Tests
 
 When you run tests from the command line, use the [--ssl](../../reference/command-line-interface.md#--ssl-options) flag to enable HTTPS on a proxy server. Specify options required to initialize [a Node.js HTTPS server](https://nodejs.org/api/https.html#https_https_createserver_options_requestlistener) after this flag in a semicolon-separated string. The most commonly used SSL options are described in the [TLS topic](https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options) in the Node.js documentation.
 
-The example below uses the PFX encoded private key and certificate chain to create an HTTPS server.
+The examples below use a previously generated `.pfx` file to establish an HTTPS server.
+
+In ***Chrome*** and ***Chromium-Based*** Browsers use `--allow-insecure-localhost` flag and [`--hostname`](../../reference/command-line-interface.md#--hostname-name) CLI parameter to make Chrome trust the certificate.
```

Comment:
This doesn't explain why is `--hostname` set to `localhost` here.

We should note that we tell TestCafe to run on `localhost` in order to be able to use browser's `--allow-insecure-localhost`. If we use a different URL for TestCafe, we should specify `--ignore-certificate-errors`.